### PR TITLE
Parse Infinity and -Infinity if JSONSL_PARSE_NAN defined

### PR DESCRIPTION
**Pull request by ajdavis**

### CODE REVIEW

**Comment by ajdavis** [comment]
Hold off on reviewing this, I think I want to make one change.

**Comment by ajdavis** [comment]
OK, this is ready for review please!

File: jsonsl.c [hunk 1]
```diff
@@ -378,7 +391,8 @@ jsonsl_feed(jsonsl_t jsn, const jsonsl_char_t *bytes, size_t nbytes)
                 goto GT_SPECIAL_NUMERIC;
             }
 
-            if (state->special_flags & JSONSL_SPECIALf_NUMERIC) {
+            if (state->special_flags & JSONSL_SPECIALf_NUMERIC &&
```
Comment (mnunberg) [inline jsonsl.c]: I'd prefer some `(parens)`.

Also, maybe ifdef this out too? - I'm guessing any compiler worth its salt should be able to optimize this out, though .. -- eg. foo&bar && !&foo&0

Comment (ajdavis) [inline jsonsl.c]: I've added the parens. I thought about ifdef'ing but I agree that the compiler will optimize this out, and it keeps the code much cleaner and easier to work on if there are fewer ifdefs.

**Comment by mnunberg** [comment]
Curious.. what are you using jsonsl for?

**Comment by ajdavis** [comment]
Hah, I should've mentioned it earlier. We recently replaced libyajl with jsonsl as our JSON parser in libbson:

http://mongoc.org/libbson/current/json.html#converting-json-to-bson
https://github.com/mongodb/libbson

**Pull request merged**